clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2031,7 +2031,8 @@ private void checkConsumptionStateWhenStart(
                 offsetLag,
                 previousOffsetLag,
                 offsetLagThreshold);
-            if (offsetLag < previousOffsetLag + offsetLagDeltaRelaxFactor * offsetLagThreshold) {
+            if (newPartitionConsumptionState.getReadyToServeInOffsetRecord()
```

Comment:
This means we might see deployment stuck / slowdown in next round of deploymenmt, due to the fact that certain resources were falsely marked as RTS with lag right? Although I think it is actually a good catch and it should only affect one time.